File: src/expansion/scala/mrtjp/projectred/expansion/block/ChargingBenchBlock.scala
package mrtjp.projectred.expansion.block

import mrtjp.projectred.expansion.{BaseMachineBlock, TileChargingBench}
import net.minecraft.block.{Block, BlockState}
import net.minecraft.item.BlockItemUseContext
import net.minecraft.state.StateContainer

class ChargingBenchBlock extends BaseMachineBlock(() => new TileChargingBench) {

    override protected def createBlockStateDefinition(builder: StateContainer.Builder[Block, BlockState]): Unit = {
        super.createBlockStateDefinition(builder)
        builder.add(BaseMachineBlock.CHARGED_PROPERTY)
    }

    override def getStateForPlacement(context: BlockItemUseContext): BlockState = {
        super.getStateForPlacement(context)
            .setValue(BaseMachineBlock.CHARGED_PROPERTY, Boolean.box(false))
    }
}

File: src/expansion/java/mrtjp/projectred/expansion/block/ChargingBenchBlock.java
package mrtjp.projectred.expansion.block;

import codechicken.lib.vec.Rotation;
import mrtjp.projectred.core.block.ProjectRedBlock;
import mrtjp.projectred.expansion.tile.ChargingBenchTile;
import net.minecraft.block.Block;
import net.minecraft.block.BlockState;
import net.minecraft.item.BlockItemUseContext;
import net.minecraft.state.StateContainer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.IBlockReader;

import javax.annotation.Nullable;

public class ChargingBenchBlock extends ProjectRedBlock {

    public ChargingBenchBlock() {
        super(STONE_PROPERTIES);
    }

    @Override
    protected TileEntity createTileEntityInstance(BlockState state, IBlockReader world) {
        return new ChargingBenchTile();
    }

    @Nullable
    @Override
    public BlockState getStateForPlacement(BlockItemUseContext context) {
        int r = Rotation.rotationTo(0, context.getHorizontalDirection().ordinal());
        return this.defaultBlockState()
                .setValue(CHARGED, false);
    }

    @Override
    protected void createBlockStateDefinition(StateContainer.Builder<Block, BlockState> builder) {
        builder.add(CHARGED);
    }
}
